packages/react-router/src/ReactRouter/StackManager.tsx

```diff
@@ -97,16 +99,64 @@ export class StackManager extends React.PureComponent<StackManagerProps, StackMa
         this.ionRouterOutlet?.props.children,
         routeInfo
       ) as React.ReactElement;
+
       if (enteringViewItem) {
         enteringViewItem.reactElement = enteringRoute;
+      } else if (enteringRoute) {
+        enteringViewItem = this.context.createViewItem(this.id, enteringRoute, routeInfo);
+        this.context.addViewItem(enteringViewItem);
       }
-      if (!enteringViewItem) {
-        if (enteringRoute) {
-          enteringViewItem = this.context.createViewItem(this.id, enteringRoute, routeInfo);
-          this.context.addViewItem(enteringViewItem);
-        }
-      }
+
       if (enteringViewItem && enteringViewItem.ionPageElement) {
+        /**
+         * If the entering view item is the same as the leaving view item,
+         * then we don't need to transition.
+         */
+        if (enteringViewItem === leavingViewItem) {
+          /**
+           * If the entering view item is the same as the leaving view item,
+           * we are either transitioning using parameterized routes to the same view
+           * or a parent router outlet is re-rendering as a result of React props changing.
+           *
+           * If the route data does not match the current path, the parent router outlet
+           * is attempting to transition and we cancel the operation.
+           */
+          if (enteringViewItem.routeData.match.url !== routeInfo.pathname) {
+            return;
+          }
+        }
```

Comment:
Would it make sense to only check if `enteringViewItem.routeData.match.url !== routeInfo.pathname`?

If `enteringViewItem === leavingViewItem` but `enteringViewItem.routeData.match.url === routeInfo.pathname` then that would mean you went from `/page/1` to `/page/1` which I don't think is possible.

Reply:
This specific check is to catch the difference in behavior between changing between tabs vs. pushing to another route that uses the same component. 

With tabs and our reliance on React's re-rendering as a result of props changing to trigger route transition, the `/tabs` component will reach this check when navigating between `/tabs/tab1` to `/tabs/tab2`. But, since the entering view item's rout date is `/tabs` and not `/tabs/tab1`, it will return early (skipping the unnecessary transition). 

If pushing to the same detail page, the entering and leaving view item will be the same, but the entering view item's route will be `/page/1` and the route info will be `/page/1`, so the transition will occur. 

I'm a little worried about opening up this check without the entering/leaving comparison, since this is the only case that is currently attempting to transition when it shouldn't. Ideally all of this goes away, if we move off React's re-rendering/prop changes for handling route transitions and follow closer suite to the Vue implementation.

Reply:
Makes sense, let's keep it as is 👍